tests/services/test_cases.py

```diff
@@ -0,0 +1,215 @@
+            mock_case_response,
+        ]
+        for _ in cases_service.get_all(name="test-case"):
+            continue
+
+        expected_params = {
+            "name": "test-case",
+            "subject": None,
+            "assignee": None,
+            "createdAt": None,
+            "updatedAt": None,
+            "status": None,
+            "pgNum": 1,
+            "pgSize": 500,
+            "srtDir": "asc",
+            "srtKey": "number",
+        }
+        mock_connection.get.assert_called_once_with(_BASE_URI, params=expected_params)
+
+    def test_get_all_called_with_expected_url_and_all_optional_params(
+        self, mock_connection, mock_case_response
+    ):
+        cases_service = CasesService(mock_connection)
+        mock_connection.get.side_effect = [
+            mock_case_response,
+            mock_case_response,
+        ]
+        for _ in cases_service.get_all(
+            name="test-case",
+            subject="test",
+            assignee="user-uid",
+            updated_at="2010-04-30T001",
+            created_at="2010-01-03T002",
+            status="open",
+        ):
+            continue
+
+        expected_params = {
+            "name": "test-case",
+            "subject": "test",
+            "assignee": "user-uid",
+            "createdAt": "2010-01-03T002",
+            "updatedAt": "2010-04-30T001",
+            "status": "open",
+            "pgNum": 1,
+            "pgSize": 500,
+            "srtDir": "asc",
+            "srtKey": "number",
+        }
+        mock_connection.get.assert_called_once_with(_BASE_URI, params=expected_params)
+
+    def test_export_called_with_expected_url_and_params(self, mock_connection):
+        cases_service = CasesService(mock_connection)
+        cases_service.export_summary(_TEST_CASE_NUMBER)
+        assert mock_connection.get.call_args[0][0] == u"/api/v1/case/{}/export".format(
+            _TEST_CASE_NUMBER
+        )
+
+    def test_get_case_called_with_expected_url_and_params(self, mock_connection):
+        cases_service = CasesService(mock_connection)
+        cases_service.get_case(_TEST_CASE_NUMBER)
+        assert mock_connection.get.call_args[0][0] == u"/api/v1/case/{}".format(
+            _TEST_CASE_NUMBER
+        )
+
+    def test_update_called_with_expected_url_and_params(
+        self, mock_connection, mock_get_response
+    ):
+        cases_service = CasesService(mock_connection)
+        mock_connection.get.return_value = mock_get_response
+        cases_service.update(_TEST_CASE_NUMBER, findings=u"x")
+        data = {
+            "name": "string",
+            "subject": "string",
+            "assignee": "string",
+            "description": None,
+            "status": "OPEN",
+            "findings": u"x",
+        }
+        mock_connection.put.assert_called_once_with(
+            u"/api/v1/case/{}".format(_TEST_CASE_NUMBER), json=data
+        )
+
+    def test_update_failure_raised_appropriate_custom_exception(
```

Comment:
Missing a test that makes sure `update` uses current values of the case when not given new values

Reply:
The test above this test does cover that case.

Reply:
Oh I see. I guess it was not obvious based on the test signature